helix-loader/src/config.rs

```diff
@@ -19,7 +19,7 @@ pub fn user_lang_config() -> Result<toml::Value, toml::de::Error> {
         .into_iter()
         .chain([default_lang_config()].into_iter())
         .fold(toml::Value::Table(toml::value::Table::default()), |a, b| {
-            crate::merge_toml_values(b, a, true)
+            crate::merge_toml_values(b, a, 3)
```

Comment:
I think it's worth a comment here explaining why it's 3 for the languages.toml (just here is probably fine, I don't think it needs to be repeated in the tests)

Reply:
Is this better?

Reply:
Yeah, great writeup! 👍